Assets/Scripts/Tile.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Tile : MonoBehaviour
+{
+	private int _distanceFromCamera = 10;
+	
+	public int _value;
+	
+	public Vector2 homePosition; // where it will slide back to if not over a valid slot
+	public Vector2 targetPosition; // if over a valid slot, will slide to this point.
+
+	// Update is called once per frame
+	void Update ()
+	{
+		if (!_dragging)
+		{
+			slide();
+		}
+	}
+
+	private Boolean _dragging = false;
+	void OnMouseDrag()
+	{
+		_dragging = true;
+		Vector3 mousePosition = new Vector3(Input.mousePosition.x, Input.mousePosition.y, _distanceFromCamera);
+		Vector3 objectPosition = Camera.main.ScreenToWorldPoint(mousePosition);
+
+		transform.position = objectPosition;
+	}
+
+	void OnMouseUp()
+	{
+		_dragging = false;
+	}
+
+	void slide()
+	{
+		if (overTarget())
```

Comment:
Maybe set targetPosition in the if condition, setting it to homePosition if not over a target. Then just call slideToPoint outside the if statement.